pycheribuild/projects/cross/cheribsd.py

```diff
@@ -1630,50 +1636,63 @@ def get_default_kernel_abi(self):
             kernABI = KernelABI.NOCHERI
         return kernABI
 
+    def _get_config_variants(self, platforms: list, kernABIs: list, combine_flags: list, **filter_kwargs):
+        flag_values = itertools.product([True, False], repeat=len(combine_flags))
+        combine_tuples = list(itertools.product(platforms, kernABIs, flag_values))
+        configs = []
+        for platform, kernABI, flag_tuple in combine_tuples:
+            combined_filter = {flag: v for flag, v in zip(combine_flags, flag_tuple)}
+            filter_kwargs.update(combined_filter)
+            configs += CheriBSDConfigTable.get_configs(self.crosscompile_target, platform, kernABI, **filter_kwargs)
+        return configs
+
+    def _get_kABIs_to_build(self):
+        default_kABI = self.get_default_kernel_abi()
+        kernABIs = [default_kABI]
+        if self.build_alternate_abi_kernels:
+            otherABI = KernelABI.PURECAP if default_kABI != KernelABI.PURECAP else KernelABI.HYBRID
+            kernABIs.append(otherABI)
+        return kernABIs
+
+    def _get_all_kernel_configs(self):
+        kernABIs = self._get_kABIs_to_build()
+        platform = self.get_default_kernel_platform()
+        combinations = []
+        if self.build_bench_kernels:
+            combinations.append("benchmark")
+        if self.caprevoke_kernel:
+            combinations.append("caprevoke")
+        if self.build_fpga_kernels:
+            combinations.append("mfsroot")
```

Comment:
This does not seem right. Should it be like this below?
```python3
        if self.build_fpga_kernels:
            configs += self._get_config_variants([ConfigPlatform.fpga_platforms()], kernABIs, combinations + ["mfsroot"])
```

Reply:
I think you are right